model/lifecycle.go

```diff
@@ -1756,19 +1760,22 @@ func addNewTasks(ctx context.Context, activationInfo specificActivationInfo, v *
 		}
 		// update build activation status if tasks have since been activated
 		if !wasActivated && b.Activated {
-			if err := build.UpdateActivation([]string{b.Id}, true, evergreen.DefaultTaskActivator); err != nil {
-				return nil, err
-			}
+			buildIdsToActivate = append(buildIdsToActivate, b.Id)
 		}
 	}
-	if activationInfo.hasActivationTasks() {
-		grip.Error(message.WrapError(v.UpdateBuildVariants(), message.Fields{
-			"message": "unable to add batchtime tasks",
-			"version": v.Id,
-		}))
+	if len(buildIdsToActivate) > 0 {
+		if err := build.UpdateActivation(buildIdsToActivate, true, evergreen.DefaultTaskActivator); err != nil {
+			return nil, err
```

Comment:
Why do we want to switch from logging the error and continuing to returning an error?

Reply:
Are you referring to the log I removed "unable to add batchtime tasks" ? (I'm not sure why github isn't showing your comment in context on the file tab, alas) 

That was when the update _only_ handled adding batchtime info, but now with the refactor it also updates activation, which was an error before and should still be an error.

Reply:
ah that makes perfect sense, thank you!